tensorflow_addons/losses/kappa_loss.py

```diff
@@ -103,7 +112,7 @@ def __init__(
             self.weight_mat = (col_mat - row_mat) ** 2
 
     def call(self, y_true, y_pred):
-        y_true = tf.cast(y_true, dtype=self.dtype)
+        y_true = tf.cast(y_true, dtype=tf.keras.backend.floatx())
```

Comment:
Is this required?

Reply:
Test, because there is a `tf.matmul(y_true, self.col_label_vec)` down below, where `self.col_label_vec` is of type `tf.keras.backend.floatx()`.

Reply:
Ok so the direct constrain is  `dtype=self.col_label_vec.dtype`?